zerver/lib/actions.py

```diff
@@ -3667,16 +3668,76 @@ def do_invite_users(user_profile, invitee_emails, streams, body=None):
         stream_ids = [stream.id for stream in streams]
         prereg_user.streams.set(stream_ids)
 
-        event = {"email": prereg_user.email, "referrer_id": user_profile.id, "email_body": body}
-        queue_json_publish("invites", event,
-                           lambda event: do_send_confirmation_email(prereg_user, user_profile, body))
+        queue_send_invitation_email(prereg_user, user_profile, body)
 
     if skipped:
         raise InvitationError(_("Some of those addresses are already using Zulip, "
                                 "so we didn't send them an invitation. We did send "
                                 "invitations to everyone else!"),
                               skipped, sent_invitations=True)
 
+def queue_send_invitation_email(prereg_user, user_profile, body):
+    event = {"email": prereg_user.email, "referrer_id": user_profile.id, "email_body": body}
+    queue_json_publish("invites", event,
+                           lambda event: do_send_confirmation_email(prereg_user, user_profile, body))
+
+def do_get_user_invites():
+    
+    from confirmation import settings as confirmation_settings
+
+    days_to_activate = getattr(settings, 'ACCOUNT_ACTIVATION_DAYS', 7)
+    active_value = getattr(confirmation_settings, 'STATUS_ACTIVE', 1)
+
+    lowest_datetime = timezone_now() - datetime.timedelta(days=days_to_activate)
+    prereg_users = PreregistrationUser.objects.exclude(status=active_value).filter(invited_at__gte=lowest_datetime)
+    invites = []
+
+    for invitee in prereg_users:
+        invites.append(dict(email=invitee.email, ref=invitee.referred_by.email, invited=invitee.invited_at.strftime("%Y-%m-%d %H:%M:%S"), id=invitee.id))
+
+    return invites
+
+def do_revoke_user_invite(id):
+    try:
+        user = PreregistrationUser.objects.get(id__iexact=id)
+        email = user.email
+        user.delete()
+        clear_scheduled_invitation_emails(email)
+        return True
+    except PreregistrationUser.DoesNotExist:
+        return False
```

Comment:
You've got a security bug here, which is that you're not checking that the realm matches the one the user admins.  Sorta same problem as with the last function.  You should have this pass in the realm ID and throw a JsonableError if the realms don't match.

Reply:
should be fixed